### ISSUE TITLE
Handle opentracing context propagation across threads

### ISSUE THREAD

**Issue by Steve-Fontes** [eclipse/microprofile-opentracing#27]
May need to use InheritedThreadLocal, and/or a special Executor Class to make sure context is propagated correctly between threads. In opentracing 1.0 we will require the developer to add code to transfer the required information between threads.

**Comment by pavolloffay** [comment]
Agree, ScopeManager/ActiveSpanManager should be configurable in the tracer, I can propose `InheritedThreadLocal`-based impl in OpenTracing.
